Package/RedPy/redpy.py

```diff
@@ -26,146 +27,121 @@ def download(self, subreddit, number=5, sort_option=None,
             num_of_videos = number - num_of_images
         self.download_images(subreddit, num_of_images, sort_option)
         self.download_videos(subreddit, num_of_videos, sort_option)
-            
 
-    def download_images(self, subreddit, number=5, sort_option=None):
+    def download_images(self, subreddit, number=5, sort_option=''):
         """Downloads images from subreddit.
             subreddit="Name of subreddit"
             number=Number of images to be downloaded
             sort_option=new/hot/top
         """
-        subreddit = subreddit.strip('/')
-        if sort_option == None:
-            sort_option = ''
+        subreddit = subreddit.split('/')[-1] if "/" in subreddit else subreddit
 
-        self.url = 'https://www.reddit.com/r/' + subreddit + '/'+  sort_option  +  '.json'
-        print(self.url)
+        json_data = self._generateJSON(subreddit, sort_option)
```

Comment:
in `_generateJSON` function, if there is an error in the request, you return `None`. You are not handling this `None` anywhere. 
This will cause error since `json_data ` will be None and you pass this to the next functions.

We are actually handling this with the `try-except` block in the getMedia functions but there should be a better way of handling this.

Maybe raise an exception there when the request is 200. Or check if `json_data` is None and return/raise and ask the user to use another user-agent.

Perhaps raising and exception would be a better option

Reply:
Modified to raise exception in such errors